bootloaders/riotboot_dfu/doc.txt

```diff
@@ -51,7 +51,16 @@ When the device is attached and in DFU mode (or the current firmware uses the `u
 new firmware can be flashed to slot 0 using:
 
 ```
-$ FEATURES_REQUIRED+=riotboot USEMODULE+=usbus_dfu make -C examples/saul BOARD=particle-xenon PROGRAMMER=dfu-util all riotboot/flash-slot0
+$ FEATURES_REQUIRED+=riotboot USEMODULE+=usbus_dfu make -C examples/saul BOARD=particle-xenon \
+  PROGRAMMER=dfu-util USB_VID=1209 USB_PID=7d02 all riotboot/flash-slot0
+```
+
+Instead of setting `USB_VID` and `USB_PID`, the variable `DFU_USB_ID` could also
+be used to specify the DFU device to be used.
+
+```
+$ FEATURES_REQUIRED+=riotboot USEMODULE+=usbus_dfu make -C examples/saul BOARD=particle-xenon \
+  PROGRAMMER=dfu-util DFU_USB_ID=1209:7d02 all riotboot/flash-slot0
```

Comment:
When none of them are defined, the `dfu-util` command looks like:
```
dfu-util --device : --alt 0 --download
```
So the `--device` parameter seems to be malformed and is only working because `dfu-util` seems to tolerant. If there are more DFU capable devices, `DFU_USB_ID` or `USB_VID` and `USB_PID` have to be defined.

Reply:
> If there are more DFU capable devices, DFU_USB_ID or USB_VID and USB_PID have to be defined.

True, but that's not the most common setup (unless you have a weird USB based webcam integrated to the laptop which support DFU...).

Still, this is weird because `USB_VID` and `USB_PID` should have default values so `--device`  shouldn't looks like that.

Reply:
The default values of `USB_VID` and `USB_PID` are set either in `makefiles/usb-codes.inc.mk` if `CONFIG_USB_VID` and `CONFIG_USB_PID` are defined, or in the makefile of the application.The latter is usually only the case for applications that implement USB device interfaces, such as `bootloaders/riotboot_dfu` itself. Applications which use `usbus_dfu` only for downloading usually don`t define `USB_VID` and `USB_PID`.

The option we have are:
- We leave it as is with the malformed `--device` argument.
- We define the default values of `USB_VID` and `USB_PID` in `makefiles/usb-codes.inc.mk` if module `usbus_dfu` is used.

Reply:
I guess the latter is the better.